### PR TITLE
Users can purchase products

### PR DESCRIPTION
**Pull request by LevinGeorge**

### CODE REVIEW

File: app/controllers/products_controller.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class ProductsController < ApplicationController
 	before_action :authenticate_user!, except: [:index, :show]
-	before_action :set_shop
+	before_action :set_shop, except: [:checkout]
```
Comment (danielpaul) [inline app/controllers/products_controller.rb]: This seems to be when you had the checkout method here?

File: app/models/purchase.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+class Purchase < ActiveRecord::Base
+  belongs_to :product
```
Comment (danielpaul) [inline app/models/purchase.rb:2]: Also should belong_to a user

File: config/routes.rb [hunk 3]
```diff
@@ -16,7 +16,8 @@
   end
 
   get "dashboard" => "pages#dashboard", as: "dashboard"
-
+  post "checkout/:id" => "purchases#create", as: "checkout"
+  get "purchase_history" => "purchases#index"
```
Comment (danielpaul) [inline config/routes.rb]: could we have the route here as `dashboard/purchases` please. Also move this line right after the dashboard get route.

File: app/controllers/products_controller.rb (hunk 1 above)
Comment (LevinGeorge) [inline app/controllers/products_controller.rb]: Oh that was from before. I will remove the line now.

File: config/routes.rb (hunk 3 above)
Comment (LevinGeorge) [inline config/routes.rb]: Didn't understand what you meant by dashboard/purchases

Comment (danielpaul) [inline config/routes.rb]: `get 'dashboard/purchases' => 'purchases#index'`

**Comment by danielpaul** [comment]
@LevinGeorge Is this ready to be merged? I think that there is just one comment there that needs to be resolved?

**Comment by LevinGeorge** [comment]
@danielpaul  Which comment?

**Comment by starkshaw** [comment]
@danielpaul Please fix the conflict in this branch and merge it. I need it on Purchase history. Any new features related to purchase gonna need this. Please keep the `*.html.haml` files with conflict new, I'll fix it.

**Comment by danielpaul** [comment]
@LevinGeorge could you rebase from `master` to this branch and fix the merge conflicts yourself? https://github.com/SprintFire/marketplace/wiki/Rebasing-from-master-branch

**Comment by starkshaw** [comment]
```
U   app/models/user.rb
U   app/views/pages/dashboard.html.haml
U   app/views/products/show.html.haml
U   db/schema.rb
error: commit is not possible because you have unmerged files.
hint: Fix them up in the work tree, and then use 'git add/rm <file>'
hint: as appropriate to mark resolution and make a commit.
fatal: Exiting because of an unresolved conflict.
```

Here's the conflict.

**Comment by starkshaw** [comment]
@danielpaul The conflicts are resolved. Please merge then I can focus on Purchase History.

**Comment by danielpaul** [comment]
thanks @starkshaw

**Pull request merged**